src/ChainRulesTestUtils.jl

```diff
@@ -0,0 +1,186 @@
+module ChainRulesTestUtils
+
+using ChainRulesCore
+using ChainRulesCore: AbstractDifferential
```

Comment:
It'd be nice if `ChainRulesCore` would export this so that we don't need this line.

Reply:
🤔 I'm not sure if other packages should dispatch or subtype this (except for this one) ...but maybe we should export it - open an issue?

Reply:
https://github.com/JuliaDiff/ChainRulesCore.jl/issues/124